src/main/java/org/kiwiproject/dropwizard/util/health/keystore/ExpiringKeystoreHealthCheck.java

```diff
@@ -0,0 +1,141 @@
+package org.kiwiproject.dropwizard.util.health.keystore;
+
+import static java.util.Collections.emptyList;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.groupingBy;
+import static org.kiwiproject.base.KiwiStrings.format;
+
+import com.codahale.metrics.health.HealthCheck;
+import com.google.common.annotations.VisibleForTesting;
+import lombok.extern.slf4j.Slf4j;
+import org.kiwiproject.dropwizard.util.config.KeystoreConfig;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.StreamSupport;
+
+/**
+ * Health check that checks whether X.509 certificates are valid, expired, or will be expiring soon as configured
+ * int a {@link KeystoreConfig}
```

Comment:
```suggestion
 * in a {@link KeystoreConfig}.
```

(You should remember to add a period to ends of sentences... :smiley: )